apps/projects/contracts/Projects.sol

```diff
@@ -23,95 +75,224 @@ contract Projects is AragonApp {
 
     /**
-     * Remove an entry from the registry.
-     * @param _id The ID of the entry to remove
+     * @notice Remove an entry from the projects registry.
+     * @param _repoId The id of the Github repo in the projects registry
      */
     function removeRepo(
-        bytes32 _id
-    ) public auth(REMOVE_REPO_ROLE)
+        bytes32 _repoId
+    ) external auth(REMOVE_REPO_ROLE)
     {
         // Take the repo out of the repo array in constant time by replacing the element
         // with last element
-        repoIDs[repos[_id].index] = repoIDs[repoIDs.length - 1];
-        repoIDs.length = repoIDs.length - 1;
-        delete repos[_id];
-        emit RepoRemoved(_id);
+        repoIds[repos[_repoId].index] = repoIds[repoIds.length - 1];
+        repoIds.length = repoIds.length - 1;
+        delete repos[_repoId];
+        emit RepoRemoved(_repoId);
     }
 
-    function getRepoArrayLength() public view returns (uint256) {
-        return repoIDs.length;
-    }
+///////////////////
+// Bounty functions
+///////////////////
 
     /**
-     * Get an entry from the registry.
-     * @param _id The ID of the entry to get
+     * @notice accept a given fulfillment
+     * @param _repoId The id of the Github repo in the projects registry
+     * @param _issueNumber the index of the bounty
+     * @param _bountyFulfillmentId the index of the fulfillment being accepted
      */
-    function getRepo(bytes32 _id) public view returns (bytes32 _owner, bytes32 _repo) {
-        _owner = repos[_id].owner;
-        _repo = repos[_id].repo;
+    function acceptFulfillment(
+        bytes32 _repoId,
+        uint256 _issueNumber, 
+        uint _bountyFulfillmentId
+    ) external auth(ADD_BOUNTY_ROLE) 
+    {
+        GithubIssue storage issue = repos[_repoId].issues[_issueNumber];
+        bounties.acceptFulfillment(issue.standardBountyId, _bountyFulfillmentId);
+        emit FulfillmentAccepted(_repoId, _issueNumber, _bountyFulfillmentId);
     }
 
-    function addBounties(bytes32 _repoID, uint256[] _issueNumbers, uint256[] _bountySizes) public  auth(ADD_BOUNTY_ROLE) {
-        for (uint i = 0; i < _issueNumbers.length; i++) {
-            _addBounty(_repoID, _issueNumbers[i], _bountySizes[i]);
+    /**
+     * @notice add bulk bounties
+     * @param _repoId The id of the Github repo in the projects registry
+     * @param _issueNumbers an array of bounty indexes
+     * @param _bountySizes an array of bounty sizes
+     * @param _deadlines an array of bounty deadlines
+     * @param _tokenBounties an array of token bounties
+     * @param _tokenContracts an array of token contracts
+     * @param _ipfsAddresses a string of ipfs addresses
+     */
+    function addBounties(
+        bytes32 _repoId,
+        uint256[] _issueNumbers, 
+        uint256[] _bountySizes, 
+        uint256[] _deadlines,
+        bool[] _tokenBounties,
+        address[] _tokenContracts,
+        string _ipfsAddresses
+    ) public payable auth(ADD_BOUNTY_ROLE)
```

Comment:
This will be complicated to handle on the front end; we might have to use the hacky proxy approach being used in the allocations since it's currently hard to call payables from the front end.

Reply:
We should probably re-factor this to go through the vault but that's not really a short-term strategy.